server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -79,4 +80,15 @@ public final String simpleName() {
     public String toString() {
         return Strings.toString(this);
     }
+
+    private static final StringLiteralDeduplicator fieldNameStringDeduplicator = new StringLiteralDeduplicator();
+
+    /**
+     * Interns the given field name string through a {@link StringLiteralDeduplicator}.
+     * @param fieldName field name to intern
+     * @return interned field name string
+     */
+    public static String internFieldName(String fieldName) {
```

Comment:
does this need to be public?

Reply:
Right damn I extracted this from another change but in isolation here, this could've been `protected` indeed.